app/controllers/communitarian/users_controller.rb

```diff
@@ -0,0 +1,220 @@
+# frozen_string_literal: true
+
+module Communitarian
+  class UsersController < ::ApplicationController
+    requires_plugin Communitarian
+
+    before_action :respond_to_suspicious_request
+    skip_before_action :verify_authenticity_token, :redirect_to_login_if_required
+
+    def new
+      params.require(:email)
+      params.require(:username)
+      params.require(:invite_code) if SiteSetting.require_invite_code
+      params.permit(:user_fields)
+
+      unless SiteSetting.allow_new_registrations
+        return fail_with("login.new_registrations_disabled")
+      end
+
+      if params[:password] && params[:password].length > User.max_password_length
+        return fail_with("login.password_too_long")
+      end
+
+      if params[:email].length > 254 + 1 + 253
```

Comment:
It looks confusing 😕

Reply:
I simply copied the whole logic from the original UsersController. I decide leave it as is since in this case it will be more ease for us to compare the difference between `UsersController` and `Communitarian::UsersController`